### ISSUE TITLE
Fix draft and outbox APIs and data handling

### ISSUE THREAD

**Issue by st3iny** [nextcloud/mail#8631]
Followup to https://github.com/nextcloud/mail/issues/4768

The draft and outbox controllers contain a lot of duplicated code and there is no clear separation currently. Furthermore, the frontend is not API agnostic and is written with the knowledge that both APIs internally point to the same database table.

Consider the following code responsible for sending a message:
https://github.com/nextcloud/mail/blob/10162f0d5eba3e245f2939d7d15d93d1213e14e8/src/components/NewMessageModal.vue#L309-L321

This code creates a draft object by calling `DraftsController::create` and then uses the resulting ID to call `OutboxController::update` later. This is problematic as an object from one API is passed to another API. Currently, it kind of works because both APIs share the same database internally.

**Regarding #8630:** The problem is that the type of an outbox message is never set to `TYPE_OUTGOING` in `OutboxController::update`. Instead, this is handled in `DraftsController::update` (_why is the drafts controller responsible for queuing/sending messages?_). Thus, the draft is never "converted" to an outbox message and will never be shown in the outbox.

## Solution

I had a call with Anna, who designed large parts of the drafts backend. We came up with the following flow:

![message_flow_chart](https://github.com/nextcloud/mail/assets/1479486/a2602521-601b-42fc-a6bd-756af8a20a63)

#### Compose a new message and send it

1. Compose new message => Create local draft via `DraftsController`.
4. Edit message => Update is handled by `DraftsController`.
5. Press send => Draft is converted to an outbox message via `OutboxController`.
6. ...
7. Outbox message is canceled/edited via the frontend => All updates go through the `OutboxController`.

#### Compose a new message and don't send it

1. Compose new message => Create local draft via `DraftsController`.
3. Edit message => Update is handled by `DraftsController`.
4. Close composer session => Draft is converted back to an IMAP message via `DraftsController`

#### Edit an IMAP draft and send it

1. Open IMAP draft => Convert IMAP message to local draft via `DraftsController`.
2. Edit draft => Update is handled by `DraftsController`.
3. Press send => Draft is converted to an outbox message via `OutboxController`.
5. ...
6. Outbox message is canceled/edited via the frontend => All updates go through the `OutboxController`.

#### Edit an IMAP draft and don't send it

1. Open IMAP draft => Convert imap message to local draft via `DraftsController`.
2. Edit draft => Update is handled by `DraftsController`.
3. Close the composer session => Draft is converted back to an IMAP message via  `DraftsController`.

## TL;DR

Once a user sends an email by pressing the "send" button on the composer the message should be exclusively handled by the `OutboxController`. Currently, it is handled by both at the same time leading to weird edge cases, unclear responsibilities and a **lot** of duplicated code.

**Comment by ChristophWurst** [comment]
I'll try to fix this
